lib/core/src/Cardano/Pool/Metrics.hs

```diff
@@ -79,6 +90,74 @@ data Block = Block
     , producer :: PoolId
     } deriving (Eq, Show, Generic)
 
+--------------------------------------------------------------------------------
+-- StakePoolLayer
+--------------------------------------------------------------------------------
+
+-- | @StakePoolLayer@ is a thin layer ontop of the DB. It is /one/ value that
+-- can easily be passed to the API-server, where it can be used in a simple way.
+newtype StakePoolLayer m = StakePoolLayer
+      { listStakePools
+          :: ExceptT ErrListStakePools m
+             [(PoolId, (Quantity "lovelace" Word64, Quantity "block" Natural))]
+      }
+
+data ErrListStakePools
+     = ErrMetricsIsUnsynced (Quantity "percent" Percentage)
+     | ErrListStakePoolsMetricsInconsistency ErrMetricsInconsistency
+
+newStakePoolLayer
+     :: DBLayer IO
+     -> NetworkLayer IO t block
+     -> Trace IO Text
+     -> IO (StakePoolLayer IO)
+newStakePoolLayer db nl tr = do
+     return $ StakePoolLayer
+        { listStakePools = do
+            nodeTip <- withExceptT undefined $ networkTip nl
+            let nodeEpoch = nodeTip ^. #slotId . #epochNumber
+
+            distr <- liftIO $ Map.fromList <$> readStakeDistribution db nodeEpoch
+            prod <- liftIO $ count <$> readPoolProduction db nodeEpoch
+
+            when (Map.null distr || Map.null prod) $ do
+                mDbTip <- liftIO poolProductionTip
+                case mDbTip of
+                    Nothing -> throwE $ ErrMetricsIsUnsynced (Quantity minBound)
+                    Just dbTip -> do
+                        liftIO $ logInfo tr $ mconcat
+                            [ "The node tip is:\n"
+                            , pretty nodeTip
+                            , ",\nbut the last pool production stored in the db"
+                            , " is from:\n"
+                            , pretty dbTip
+                            ]
+                        let p = progress dbTip nodeTip
+                        throwE $ ErrMetricsIsUnsynced (Quantity p)
+
+            case combineMetrics distr prod of
+                Right x ->
+                    return $ Map.toList x
+                Left e ->
+                    throwE $ ErrListStakePoolsMetricsInconsistency e
+        }
+  where
+    count = Map.map (Quantity . fromIntegral . length)
+
+    poolProductionTip = readPoolProductionCursor db 1 >>= \case
+        [x] -> return $ Just x
+        _ -> return Nothing
+
+    progress :: BlockHeader -> BlockHeader -> Percentage
+    progress tip target =
```

Comment:
Why not use the progress calculation from the primitive types?

Reply:
1. This `blockHeight`-calculation would return 100% when we are at the node tip. I.e. when the `stake-pool list` results become available. `syncProgress` compare to the current time. If the node tip is behind the current time, the progress might then jump from 50% to, show results.
2. This `blockHeight` approach is simple and gives us (seemingly) perfectly linear increase in progress.
3. `syncProgress` returns `SyncProgress = Ready | Restoring`, which we don't want.
4. `syncProgress` has a `tolerance` which we don't want. Here we _know_ that we are not in sync, and want to show the _exact_ progress to the user. Even if that would happen to be 99% or 100%.